File: app/src/main/java/cn/devcxl/photosync/data/PhotoDao.java
package cn.devcxl.photosync.data;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import cn.devcxl.photosync.data.entity.PhotoEntity;
import kotlinx.coroutines.flow.Flow;

import java.util.List;

@Dao
public interface PhotoDao {
    @Query("SELECT * FROM photos ORDER BY createdAt ASC")
    Flow<List<PhotoEntity>> getAllFlow();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    Long insert(PhotoEntity entity);

    @Query("DELETE FROM photos")
    void clear();
}

File: app/src/main/java/cn/devcxl/photosync/data/PhotoDao.kt
package cn.devcxl.photosync.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import cn.devcxl.photosync.data.entity.PhotoEntity
import kotlinx.coroutines.flow.Flow

import java.util.List;

/**
 * DAO for photo entities.
 */
@Dao
interface PhotoDao {
    @Query("SELECT * FROM photos ORDER BY createdAt ASC")
    fun getAllFlow(): Flow<List<PhotoEntity>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(entity: PhotoEntity): Long

    @Query("DELETE FROM photos")
    suspend fun clear()
}